### ISSUE TITLE
add a kidectl create command to create an IdlingResource

### ISSUE THREAD

**Issue by orphaner** [kidle-dev/kidle#21]
Add a new command `create` on `kidlectl`. This command creates an `IdlingResource` in a fancy way.
The CLI flags has to be defined.

- [ ] add the core create command
- [ ] add the flags for the cronjob strategy

**Comment by doniacld** [comment]
Hi,
I would be interested to resolve my first oss issue in the hacktoberfest frame !
If you want to assign it to me ?

**Comment by orphaner** [comment]
hey, it would be great. Before starting to code, what is your suggestion of the CLI flags?

**Comment by doniacld** [comment]
Nice!

I wonder do you need flags for all the fields mentioned in the demo deploy file ?
Could you list the necessary/optional parameters and maybe default values for example `idle=false` at the creation ?

```
   1 + │ apiVersion: kidle.kidle.dev/v1beta1                                                                                   
   2 + │ kind: IdlingResource                                                                                                  
   3 + │ metadata:                                                                                                             
   4 + │   namespace: kidle-demo                                                                                               
   5 + │   name: podinfo                                                                                                       
   6 + │ spec:                                                                                                                 
   7 + │   # Reference of the workload to idle                                                                                 
   8 + │   idlingResourceRef:                                                                                                  
   9 + │     apiVersion: apps/v1                                                                                               
  10 + │     kind: Deployment                                                                                                  
  11 + │     name: podinfo                                                                                                     
  12 + │                                                                                                                       
  13 + │   # Idle or not?                                                                                                      
  14 + │   idle: false    
  ``` 
I imagine something like this: 
`kidlectl create -n podinfo -ns kidle-demo -rk Deployment -rn podinfo -i true` 

`--name` or `-n` for `name`
`--namespace` or `-ns` for `namespace`
`--idle` or `-i` for `idle`
`--reference-kind` or `-rk` for `reference kind` (I am not sur about this rk, maybe `ref.k` ? )
`--reference-name` or `-rn` for `reference name` (same as the previous one)


What do you think ?

**Comment by orphaner** [comment]
the 2 other commands of `kidlectl` declares:
- `name` as a command argument,
- `--namespace` as an optional flag.

we definitively need to be consistent on that 2 points.

Regarding the reference I suggest we use a well known standard on the `kubectl` command: a single flag named `ref` that takes a value formatted as `kind/name`; in the example the value is `deployment/podinfo`. It would be great to be able to use kind in its different forms: singular, plural and short (deployment, deployments, deploy). There has to be a way to perform the validation by calling the kubernetes apis through the client-go. I'll have a look into the `kubectl` code to find out how they do.

The `idle` flag is optional in the CRD, let's go this way on the CLI.

```
 ./cmd/kidlectl/bin/kidlectl create -h
Usage:
  kidlectl [OPTIONS] create [create-OPTIONS] Name

Application Options:
      --kubeconfig=    path to Kubernetes config file [$KUBECONFIG]

Help Options:
  -h, --help           Show this help message

[idle command options]
      -n, --namespace= IdlingResource namespace [$NAMESPACE]
      -i, --idle= The desired state of idling. Defaults to false.
      -r, --ref= The reference to the idle-able workload

[idle command arguments]
  Name:                idling resource name to create
```

We may wonder about the [cronjob strategy](https://github.com/kidle-dev/kidle/blob/main/docs/userguide.md#cronjob-idle-strategy). I suggest we add 2 optional flags:
```
      --idle-cronjob= The idle time in Cron format
      --wakeup-cronjob= The wakeup time in Cron format
```
It can be done in a second iteration.

A technical point we can do during this feature is to split the [implementation of the commands](https://github.com/kidle-dev/kidle/blob/main/cmd/kidlectl/main.go#L62) into a `cmd` package with a file per command. The main file really needs to stay light and focus on the commands declaration, the parsing of the args/flags and running the proper command. 
While doing this split, the [kidleclient.go](https://github.com/kidle-dev/kidle/blob/main/cmd/kidlectl/kidleclient.go) file has to be moved into the `cmd` package, or into its own package. What do you think?

**Comment by doniacld** [comment]
I like the idea of staying consistent with kubectl concerning the flags ! 
I am okay to postpone the cronjob strategy in the a second pr. :)

I will definitely move each command implementation in a dedicated file and even in a dedicated package and though add unit tests.
NB: Also at some point you might want to move them in the `pkg` repo if we refer to [kubectl cmd](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/cmd.go) arborescence package. 
`kidleclient.go` should live in its own package `client` at the root of `kidle`.

**Comment by orphaner** [comment]
for now, let's keep things simple with a `pkg` package in the `cmd/kidlectl` directory and a file per command. We'll discuss about separate packages per command when the number of command will grow up.

> FYI I'm AFK until monday

**Comment by doniacld** [comment]
Ok ! I do understand.
I am also AFK from this evening to tomorrow.
If you want to assign it to me I would be more than happy to begin to work on today during my spare time :)

**Comment by orphaner** [comment]
now you are assigned. Can you follow this https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork and open the permission for the maintainer to commit onto your fork? thx

**Comment by doniacld** [comment]
Thank you.
Sure! 👍🏽

**Comment by doniacld** [comment]
Just to let you know, I implemented the core of the create command and the package refactoring.
After looking in kubectl code, I could not find how the kind validation is performed. If you have any hint, please let me know.

Before opening a pr, it would be nice to add this validation and the optional flags.

**Comment by doniacld** [comment]
@orphaner I  can open a first PR and we can iterate on this base if you want to ?

**Comment by orphaner** [comment]
@doniacld sorry for the delay on that. I had a hard past week at work and this week-end I've played building lego with my little boy :)

I'm looking for the validation part;

I've created a slack channel so that we could discuss more easily: https://kidle-dev.slack.com/archives/C02JXP2JTK2

**Comment by orphaner** [comment]
@doniacld yes, you can create the PR even if it's a work in progress; this way I can easily checkout & test your branch

**Comment by orphaner** [comment]
@doniacld please get a look at this commit: https://github.com/kidle-dev/kidle/commit/81c280bc325182f9f7cfe9674d916723275b1444

**Comment by doniacld** [comment]
> @doniacld sorry for the delay on that. I had a hard past week at work and this week-end I've played building lego with my little boy :)
> 
> I'm looking for the validation part;
> 
> I've created a slack channel so that we could discuss more easily: https://kidle-dev.slack.com/archives/C02JXP2JTK2

Hey!
No worries, I will push my code this evening and connect to the slack channel! :)

**Comment by doniacld** [comment]
> @doniacld sorry for the delay on that. I had a hard past week at work and this week-end I've played building lego with my little boy :)
> 
> I'm looking for the validation part;
> 
> I've created a slack channel so that we could discuss more easily: https://kidle-dev.slack.com/archives/C02JXP2JTK2

Could you add me to the slack channel ? My email address is: donia.cld@gmail.com

**Comment by doniacld** [comment]
@orphaner 
I just opened a PR! 
And I checked your commit for the validation ! Thanks a lot :)
If you want to merge it, I will integrate it in the refactoring and add the validation in the createCommand.
Also I have a problem for the installation, I really need to fix it to test the code...
